core/admin/port_management.py

```
@@ -0,0 +1,449 @@
+
+
+def get_server_ip() -> str:
+    """Get server IP address"""
+    server_ip = os.getenv("SERVER_IP", "").strip()
+    
+    if not server_ip or server_ip == "auto":
+        # Try to detect server IP
+        try:
+            # Get default route interface
+            result = subprocess.run(
+                ["ip", "route", "get", "1.1.1.1"],
```

Ruff (0.14.10)

Starting a process with a partial executable path

(S607)